// angular-image-slider/bundles/angular-image-slider.umd.js
(function (global, factory) {
	typeof exports === 'object' && typeof module !== 'undefined' ? factory(exports, require('@angular/core'), require('@angular/animations'), require('@angular/common')) :
	typeof define === 'function' && define.amd ? define(['exports', '@angular/core', '@angular/animations', '@angular/common'], factory) :
	(factory((global['angular-image-slider'] = {}),global.ng.core,global.ng.animations,global.ng.common));
}(this, (function (exports,core,animations,common) { 'use strict';

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
var SliderComponent = (function () {
    function SliderComponent() {
        this.state = 'void';
        this.disableSliderButtons = false;
    }
    /**
     * @return {?}
     */
    SliderComponent.prototype.ngOnInit = function () {
        this.imageUrls = this.images;
    };
    /**
     * @param {?} arr
     * @param {?} reverse
     * @return {?}
     */
    SliderComponent.prototype.imageRotate = function (arr, reverse) {
        if (reverse)
            arr.unshift(arr.pop());
        else
            arr.push(arr.shift());
        return arr;
    };
    /**
     * @return {?}
     */
    SliderComponent.prototype.moveLeft = function () {
        if (this.disableSliderButtons) {
            return;
        }
        this.state = 'right';
        this.imageRotate(this.imageUrls, true);
    };
    /**
     * @return {?}
     */
    SliderComponent.prototype.moveRight = function () {
        if (this.disableSliderButtons) {
            return;
        }
        this.state = 'left';
        this.imageRotate(this.imageUrls, false);
    };
    /**
     * @param {?} $event
     * @return {?}
     */
    SliderComponent.prototype.onFinish = function ($event) {
        this.state = 'void';
        this.disableSliderButtons = false;
    };
    /**
     * @param {?} $event
     * @return {?}
     */
    SliderComponent.prototype.onStart = function ($event) {
        this.disableSliderButtons = true;
    };
    return SliderComponent;
}());
SliderComponent.decorators = [
    { type: core.Component, args: [{
                selector: 'angular-image-slider',
                template: "<div id=\"container\">\n  <div class=\"overlay-left\"></div>\n  <div class=\"overlay-right\"></div>\n  <div id=\"grid\" class=\"image-cropper\">\n    <img *ngFor=\"let url of imageUrls; let i = index\"\n      [@move]=\"state\"\n      (@move.done)=\"onFinish($event)\"\n      (@move.start)=\"onStart($event)\"\n      [attr.class]=\"i\"\n      [src]=\"url\"\n      alt=\"\" />\n  </div>\n  <div>\n    <a id=\"slider-buttons-left\" (click)=\"moveLeft()\">&#10094;</a>\n    <a id=\"slider-buttons-right\" (click)=\"moveRight()\">&#10095;</a>\n  </div>\n</div>",
                styles: ["#container{\n   position:relative;\n   overflow-x:hidden;\n}\n#grid{\n  display:grid;\n  height:150;\n  grid-template-columns:repeat(3, 1fr);\n  grid-template-rows:300px;\n  grid-column-gap:20px;\n}\n.overlay-left{\n    position:absolute;\n    display:inline-block;\n    display:grid;\n    width:7rem;\n    height:100%;\n    top:0%;\n    left:0;\n    bottom:0;\n    background-color:white;\n    z-index:1;\n    cursor:pointer;\n}\n.overlay-right{\n    position:absolute;\n    display:inline-block;\n    display:grid;\n    width:7rem;\n    height:100%;\n    top:0%;\n    right:0;\n    bottom:0;\n    background-color:white;\n    z-index:1;\n    cursor:pointer;\n}\n#grid > div{\n  background-color:gray;\n}\n#slider-buttons-left{\n  position:absolute;\n  z-index:1000;\n  top:130px;\n  float:left;\n  cursor:pointer;\n  font-size:250%;\n  color:#778899;\n  left:60px;\n}\n#slider-buttons-right{\n  position:absolute;\n  top:130px;\n  z-index:1000;\n  float:right;\n  cursor:pointer;\n  font-size:250%;\n  color:#778899;\n  right:60px;\n}\n.image-cropper{\n    height:300px;\n    display:-webkit-box;\n    display:-ms-flexbox;\n    display:flex;\n    -webkit-box-orient:vertical;\n    -webkit-box-direction:normal;\n        -ms-flex-direction:column;\n            flex-direction:column;\n    -webkit-box-pack:center;\n        -ms-flex-pack:center;\n            justify-content:center;\n    -webkit-box-align:center;\n        -ms-flex-align:center;\n            align-items:center;\n}\n.image-cropper img{\n    max-width:100%;\n    max-height:100%;\n    display:block;\n    margin:0 auto;\n}"],
                animations: [
                    animations.trigger('move', [
                        animations.state('in', animations.style({ transform: 'translateX(0)' })),
                        animations.transition('void => left', [
                            animations.style({ transform: 'translateX(100%)' }),
                            animations.animate(200)
                        ]),
                        animations.transition('left => void', [
                            animations.animate(200, animations.style({ transform: 'translateX(0)' }))
                        ]),
                        animations.transition('void => right', [
                            animations.style({ transform: 'translateX(-100%)' }),
                            animations.animate(200)
                        ]),
                        animations.transition('right => void', [
                            animations.animate(200, animations.style({ transform: 'translateX(0)' }))
                        ])
                    ])
                ]
            },] },
];
/** @nocollapse */
SliderComponent.ctorParameters = function () { return []; };
SliderComponent.propDecorators = {
    "images": [{ type: core.Input },],
};
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
var SliderModule = (function () {
    function SliderModule() {
    }
    return SliderModule;
}());
SliderModule.decorators = [
    { type: core.NgModule, args: [{
                imports: [
                    common.CommonModule
                ],
                declarations: [
                    SliderComponent
                ],
                exports: [SliderComponent]
            },] },
];
/** @nocollapse */
SliderModule.ctorParameters = function () { return []; };

exports.SliderModule = SliderModule;
exports.ɵa = SliderComponent;

Object.defineProperty(exports, '__esModule', { value: true });

})));
