// angular-image-slider/esm5/angular-image-slider.js
import { Component, Input, NgModule } from '@angular/core';
import { animate, state, style, transition, trigger } from '@angular/animations';
import { CommonModule } from '@angular/common';
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
var SliderComponent = (function () {
    function SliderComponent() {
        this.state = 'void';
        this.disableSliderButtons = false;
    }
    /**
     * @return {?}
     */
    SliderComponent.prototype.ngOnInit = function () {
        this.imageUrls = this.images;
    };
    /**
     * @param {?} arr
     * @param {?} reverse
     * @return {?}
     */
    SliderComponent.prototype.imageRotate = function (arr, reverse) {
        if (reverse)
            arr.unshift(arr.pop());
        else
            arr.push(arr.shift());
        return arr;
    };
    /**
     * @return {?}
     */
    SliderComponent.prototype.moveLeft = function () {
        if (this.disableSliderButtons) {
            return;
        }
        this.state = 'right';
        this.imageRotate(this.imageUrls, true);
    };
    /**
     * @return {?}
     */
    SliderComponent.prototype.moveRight = function () {
        if (this.disableSliderButtons) {
            return;
        }
        this.state = 'left';
        this.imageRotate(this.imageUrls, false);
    };
    /**
     * @param {?} $event
     * @return {?}
     */
    SliderComponent.prototype.onFinish = function ($event) {
        this.state = 'void';
        this.disableSliderButtons = false;
    };
    /**
     * @param {?} $event
     * @return {?}
     */
    SliderComponent.prototype.onStart = function ($event) {
        this.disableSliderButtons = true;
    };
    return SliderComponent;
}());
SliderComponent.decorators = [
    { type: Component, args: [{
                selector: 'angular-image-slider',
                template: "<div id=\"container\">\n  <div class=\"overlay-left\"></div>\n  <div class=\"overlay-right\"></div>\n  <div id=\"grid\" class=\"image-cropper\">\n    <img *ngFor=\"let url of imageUrls; let i = index\"\n      [@move]=\"state\"\n      (@move.done)=\"onFinish($event)\"\n      (@move.start)=\"onStart($event)\"\n      [attr.class]=\"i\"\n      [src]=\"url\"\n      alt=\"\" />\n  </div>\n  <div>\n    <a id=\"slider-buttons-left\" (click)=\"moveLeft()\">&#10094;</a>\n    <a id=\"slider-buttons-right\" (click)=\"moveRight()\">&#10095;</a>\n  </div>\n</div>",
                styles: ["#container{\n   position:relative;\n   overflow-x:hidden;\n}\n#grid{\n  display:grid;\n  height:150;\n  grid-template-columns:repeat(3, 1fr);\n  grid-template-rows:300px;\n  grid-column-gap:20px;\n}\n.overlay-left{\n    position:absolute;\n    display:inline-block;\n    display:grid;\n    width:7rem;\n    height:100%;\n    top:0%;\n    left:0;\n    bottom:0;\n    background-color:white;\n    z-index:1;\n    cursor:pointer;\n}\n.overlay-right{\n    position:absolute;\n    display:inline-block;\n    display:grid;\n    width:7rem;\n    height:100%;\n    top:0%;\n    right:0;\n    bottom:0;\n    background-color:white;\n    z-index:1;\n    cursor:pointer;\n}\n#grid > div{\n  background-color:gray;\n}\n#slider-buttons-left{\n  position:absolute;\n  z-index:1000;\n  top:130px;\n  float:left;\n  cursor:pointer;\n  font-size:250%;\n  color:#778899;\n  left:60px;\n}\n#slider-buttons-right{\n  position:absolute;\n  top:130px;\n  z-index:1000;\n  float:right;\n  cursor:pointer;\n  font-size:250%;\n  color:#778899;\n  right:60px;\n}\n.image-cropper{\n    height:300px;\n    display:-webkit-box;\n    display:-ms-flexbox;\n    display:flex;\n    -webkit-box-orient:vertical;\n    -webkit-box-direction:normal;\n        -ms-flex-direction:column;\n            flex-direction:column;\n    -webkit-box-pack:center;\n        -ms-flex-pack:center;\n            justify-content:center;\n    -webkit-box-align:center;\n        -ms-flex-align:center;\n            align-items:center;\n}\n.image-cropper img{\n    max-width:100%;\n    max-height:100%;\n    display:block;\n    margin:0 auto;\n}"],
                animations: [
                    trigger('move', [
                        state('in', style({ transform: 'translateX(0)' })),
                        transition('void => left', [
                            style({ transform: 'translateX(100%)' }),
                            animate(200)
                        ]),
                        transition('left => void', [
                            animate(200, style({ transform: 'translateX(0)' }))
                        ]),
                        transition('void => right', [
                            style({ transform: 'translateX(-100%)' }),
                            animate(200)
                        ]),
                        transition('right => void', [
                            animate(200, style({ transform: 'translateX(0)' }))
                        ])
                    ])
                ]
            },] },
];
/** @nocollapse */
SliderComponent.ctorParameters = function () { return []; };
SliderComponent.propDecorators = {
    "images": [{ type: Input },],
};
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
var SliderModule = (function () {
    function SliderModule() {
    }
    return SliderModule;
}());
SliderModule.decorators = [
    { type: NgModule, args: [{
                imports: [
                    CommonModule
                ],
                declarations: [
                    SliderComponent
                ],
                exports: [SliderComponent]
            },] },
];
/** @nocollapse */
SliderModule.ctorParameters = function () { return []; };
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
/**
 * Generated bundle index. Do not edit.
 */
export { SliderModule, SliderComponent as ɵa };
